packages/react-dom/src/client/ReactDOMComponent.js

```diff
@@ -9,10 +9,15 @@
 
 // TODO: direct imports like some-package/src/* are bad. Fix me.
 import {getCurrentFiberOwnerNameInDevOrNull} from 'react-reconciler/src/ReactCurrentFiber';
+import {
+  didNotMatchHydratedChildrenPropValue,
+  didNotMatchHydratedPropValue,
+  didNotMatchHydratedPropsHostInstanceHasExtraAttributes,
+  SUPPRESS_HYDRATION_WARNING,
+} from 'react-reconciler/src/ReactFiberHydrationWarning';
```

Comment:
Here `react-dom` `ReactDOMComponent` imports from `react-reconciler` to issue specific warnings not covered within the reconciler itself.